src/hugr/rewrite/simple_replace.rs

```diff
@@ -121,8 +124,7 @@ impl Rewrite for SimpleReplacement {
                 let (rem_inp_pred_node, rem_inp_pred_port) = h
                     .linked_ports(*rem_inp_node, *rem_inp_port)
                     .exactly_one()
-                    .ok() // PortLinks does not implement Debug
-                    .unwrap();
+                    .map_err(|_| SimpleReplacementError::InvalidBoundary())?;
```

Comment:
Since the maps were checked on construction, and an in-port must have exactly one linked port in any valid hugr, isn't it OK to panic here?